presto-native-execution/presto_cpp/main/tests/DynamicFunctionRegistererTest.cpp

```diff
@@ -0,0 +1,214 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include "presto_cpp/main/dynamic_registry/DynamicFunctionConfigRegisterer.h"
+#include "presto_cpp/main/dynamic_registry/DynamicFunctionRegistrar.h"
+#include "presto_cpp/main/dynamic_registry/DynamicSignalHandler.h"
+#include "velox/common/base/Fs.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/exec/tests/utils/TempFilePath.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/tests/RegistryTestUtil.h"
+#include "velox/type/SimpleFunctionApi.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#ifndef DYNAMIC_LIB_FILE_EXT
+#error DYNAMIC_LIB_FILE_EXT must be defined in CMAKE
+#endif
+const char* kFileExtension = DYNAMIC_LIB_FILE_EXT;
+
+using namespace facebook::velox;
+namespace facebook::presto::test {
+namespace {
+class DynamicFunctionRegistererTest : public testing::Test {
+ public:
+  template <typename T>
+  struct TestDynamicFunctionMock {
+    VELOX_DEFINE_FUNCTION_TYPES(T);
+    FOLLY_ALWAYS_INLINE bool call(out_type<bool>& result) {
+      result = "123";
+      return true;
+    }
+  };
+  template <typename T>
+  struct TestDynamicFunctionMock2 {
+    VELOX_DEFINE_FUNCTION_TYPES(T);
+    FOLLY_ALWAYS_INLINE bool call(
+        out_type<Varchar>& result,
+        const arg_type<Varchar>& in) {
+      result = in;
+      return true;
+    }
+  };
+  void UdfBoolDynamicFnRegistration(
+      std::string fn_name,
+      std::string nameSpace = "") {
+    facebook::presto::registerPrestoFunction<TestDynamicFunctionMock, bool>(
+        fn_name.c_str(), nameSpace.c_str());
+  }
+  void UdfVarcharDynamicFnRegistration(
+      std::string fn_name,
+      std::string nameSpace = "") {
+    facebook::presto::registerPrestoFunction<
+        TestDynamicFunctionMock2,
+        facebook::velox::Varchar,
+        facebook::velox::Varchar>(fn_name.c_str(), nameSpace.c_str());
+  }
+
+  void writeToFile(const std::string& filePath, std::string_view data) {
+    std::ofstream outputFile(filePath, std::ofstream::binary);
+    outputFile.write(data.data(), data.size());
+    outputFile.close();
+  }
+
+ private:
+  std::shared_ptr<SystemConfig> systemConfig;
+};
+
+TEST_F(DynamicFunctionRegistererTest, twoFunctions) {
+  std::string_view json = R"(
+  {
+    "dynamicLibrariesUdfMap": {
+      "tmp": {
+        "mock1": [
+          {
+            "entrypoint": "registry123",
+            "fileName" : "test",
+            "outputType": "boolean",
+            "paramTypes": [
+            ]
+          }
+        ],
+        "mock2": [
+          {
+            "fileName" : "test",
+            "outputType": "varchar",
+            "paramTypes": [
+              "varchar"
+            ]
+          }
+        ]
+      }
+    }
+  })";
+
+  // Emulate user provided config file.
```

Comment:
There is a some repetition of code between this and the test below. Can you make common functions for reuse ?